pkg/acl/acl.go

```diff
@@ -0,0 +1,230 @@
+/*
+ * JuiceFS, Copyright 2024 Juicedata, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package acl
+
+import (
+	"fmt"
+	"hash/crc32"
+	"sort"
+	"strings"
+
+	"github.com/juicedata/juicefs/pkg/utils"
+)
+
+const Version uint8 = 2
+
+type Entry struct {
+	Id   uint32
+	Perm uint16
+}
+
+type Entries []Entry
+
+func (es *Entries) Len() int           { return len(*es) }
+func (es *Entries) Less(i, j int) bool { return (*es)[i].Id < (*es)[j].Id }
+func (es *Entries) Swap(i, j int)      { (*es)[i], (*es)[j] = (*es)[j], (*es)[i] }
+
+func (es *Entries) IsEqual(other *Entries) bool {
+	if es.Len() != other.Len() {
+		return false
+	}
+	sort.Sort(es)
```

Comment:
We can assume it's ordered, call sort for newly created entries.

Reply:
It we don't sort them for checksum, then it's useless here.

It's OK to don't sort them at all.